### PR TITLE
[DUOS-1806][risk=no] fix actions on new researcher console

### PR DESCRIPTION
**Pull request by shaemarks**
[Link to Jira ticket](https://broadworkbench.atlassian.net/browse/DUOS-1806)
Summary of changes:
- Use `ResearcherActions` instead of previous action function for `new_researcher_console`
- Rename "resubmit" to "revise" 
- Remove collection from `DarCollectionTable` when revise button is clicked 
- Add collection to DAR Draft Table when revise button is clicked
- Adjust styling of buttons to be closer to `DarCollectionTable` on other consoles
- Link review button to dar application page with disabled fields 
- Adjust logic for showing researcher action buttons: 
   - cancel button is visible when the collection has no elections on it and not all dars are canceled
   - revise button is visible when all dars in the collection are canceled 

Note:
- Since the collection is split into separate dars with different reference ids, the review button currently redirects the user to the dar application page of an arbitrary dar in that collection

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by rushtong** [comment]
Style request, can we align the buttons to the left? The current centering makes it look tree-like:

Screen Shot 2022-05-18 at 6 44 15 AM

**Comment by rushtong** [comment]
Wording request - can we align the modal language with the button language when clicking "Revise"?

Screen Shot 2022-05-18 at 6 44 47 AM

**Comment by rushtong** [comment]
The one example I have for revise (DAR-138) is throwing a 500 on the back end ... need to dig into that to be sure there isn't something wrong with the API call from the front end. It's likely fine, problem is 99% likely an API issue.

**Comment by rushtong** [comment]
> The one example I have for revise (DAR-138) is throwing a 500 on the back end ... need to dig into that to be sure there isn't something wrong with the API call from the front end. It's likely fine, problem is 99% likely an API issue.

Submitted https://github.com/DataBiosphere/consent/pull/1579 to get past the 500 error I saw. The revise feature works correctly with these PRs in tandem.

File: src/components/dar_collection_table/DarCollectionTable.js [hunk 1]
```diff
@@ -206,7 +203,7 @@ export const DarCollectionTable = function DarCollectionTable(props) {
   //cancel, resubmit, and open need to be assigned as an "updateCollection" when relevant?
   //  - depends, if cancel and resubmit are locked behind modals then I only would have to pass in openCollection (only for admin and chair)
```
Comment (rushtong) [inline src/components/dar_collection_table/DarCollectionTable.js]: Are these comments still relevant? I'm not sure I understand what they're trying to get at here.

File: src/pages/NewResearcherConsole.js [hunk 2]
```diff
@@ -120,6 +119,25 @@ export default function NewResearcherConsole(props) {
...
+        head
+      )(darCollection);
+      if (isNil(referenceId)) {
+        throw new Error("Error: Could not find target collection");
+      }
+      history.push(`/dar_application/${referenceId}`);
+    } catch (error) {
+      Notifications.showError({
+        text: 'Error: Cannot view target collection'
+      });
```
Comment (rushtong) [inline src/pages/NewResearcherConsole.js]: nit: language. Can we refer to collections as Data Access Requests instead?

**Review by rushtong — CHANGES_REQUESTED** [review]
See comments inline. Functionality works great - I just think we need a couple minor visual updates.

**Comment by shaemarks** [comment]
> Wording request - can we align the modal language with the button language when clicking "Revise"?
> 
> Screen Shot 2022-05-18 at 6 44 47 AM

Should just the user-facing wording on the modal change, or should I refactor so that this action is consistently referred to as "revise" instead of "resubmit" in the code?

File: src/pages/NewResearcherConsole.js (hunk 2 above)
Comment (shaemarks) [reply src/pages/NewResearcherConsole.js]: I replaced "collection" with "Data Access Request" for messages similar to the ones above. The current success messages say something along the lines of "Revising collection DAR-123". For these messages, would be it best to: 
1) keep as "Revising collection DAR-123"
2) change to "Revising Data Access Request DAR-123"
3) change to "Revising DAR-123"

Comment (rushtong) [reply src/pages/NewResearcherConsole.js]: Option 2 looks best to me

**Comment by rushtong** [comment]
> > Wording request - can we align the modal language with the button language when clicking "Revise"?
> > Screen Shot 2022-05-18 at 6 44 47 AM
> 
> Should just the user-facing wording on the modal change, or should I refactor so that this action is consistently referred to as "revise" instead of "resubmit" in the code?

`should I refactor so that this action is consistently referred to as "revise" instead of "resubmit"` seems like the best choice to me.

**Review by quazi-h — APPROVED** [review]
Functionality is working as expected - DARs with complete elections only have the "Review" action available instead of "Cancel". Agree with Greg's suggestions on visual changes.

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Pull request merged**